### PR TITLE
Story1

### PR DESCRIPTION
**Pull request by mgallagher56**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+function connectDb(): PDO {
+    $db = new PDO("mysql:host=db; dbname=marcCollection","root","password");
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE,PDO::FETCH_ASSOC);
+    return $db;
+}
+
+
+/**
+ *Uses database connection to select field names from db and return the data as an associative array
+ * @param PDO $db PDO Database connection
```
Comment (charliecog) [inline functions.php]: line break above param

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ *
+ * @return mixed returns the results from database extraction as an associative array
+ */
+function getHeadingsFromDb(PDO $db): array
```
Comment (charliecog) [inline functions.php]: Easier to just hard code in the front end

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+
+/**
+ * Uses data collection from db to output array key as a list item
+ * @param $collectionArr array returned from db
+ *
+ * @return string if items in array <= 5, returns html list of db fields
+ */
+function outputFieldAsHeader(array $collectionArr) :string
```
Comment (charliecog) [inline functions.php]: again hard coding to the front end

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+
+/**
+ * A nested foreach loop to output the values of a multidimensional array as html list items
+ * @param array $collectionArr
+ *
+ * @return string multiple if items in array <= 5, returns html list of db items
+ */
+function outputDataAsRows(array $collectionData) :string
+{   if (count($collectionData[0]) <= 5) {
```
Comment (charliecog) [inline functions.php]: curly brace on the line above

File: functions.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    $result = '';
+    foreach ($collectionData as $name => $value) {
+        $result .= '<ul class="dataList">';
+        foreach ($value as $item) {
```
Comment (charliecog) [inline functions.php]: nested loops is more complicated than you need for this build - I have an example to show you.

File: index.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+require_once 'functions.php';
+
+$db = connectDb();
+
+$collectionArr = getHeadingsFromDb($db);
```
Comment (charliecog) [inline index.php]: just hardcode

File: index.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+<h1>Marc's SuperFun Shoe Collection</h1>
+<div>
+    <?php
+    echo outputFieldAsHeader($collectionArr);
+    ?>
+</div>
+
+<div>
+    <?php
+    echo outputDataAsRows($collectionData);
```
Comment (charliecog) [inline index.php]: pop it all on the same line and make it just the final variable here, move the function call to up with the others ```<?php echo $output; ?>```

File: marcCollection_2019-09-23.sql [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+# ------------------------------------------------------------
+
+DROP TABLE IF EXISTS `Shoes`;
+
+CREATE TABLE `Shoes` (
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `Name` varchar(255) NOT NULL DEFAULT '',
+  `Brand` varchar(255) NOT NULL DEFAULT '',
+  `Primary Colour` varchar(255) NOT NULL DEFAULT '',
+  `Release Year` year(4) NOT NULL,
```
Comment (charliecog) [inline marcCollection_2019-09-23.sql]: best practice is no spaces in table names and camelCase

File: functions.php [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            $result .= '<li class="data">' . $item . '</li>';
+        }
+        $result .= '<br>' . '</ul>';
```
Comment (charliecog) [inline functions.php]: use css instead of br tags

File: functions.php [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    };
+    return $result;}
```
Comment (charliecog) [inline functions.php]: curly brace in a weird place

File: functions.php [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    else {
+        return 'invalid input';
+    }
+
```
Comment (charliecog) [inline functions.php]: remove line break

**Review by charliecog — COMMENTED** [review]
looks good, almost there, just a couple of bits to mop up. nice work!

File: functions.php (hunk 1 above)
Comment (mgallagher56) [reply functions.php]: Fixed

File: functions.php (hunk 2 above)
Comment (mgallagher56) [reply functions.php]: Fixed

File: functions.php (hunk 3 above)
Comment (mgallagher56) [reply functions.php]: Fixed

File: functions.php (hunk 4 above)
Comment (mgallagher56) [reply functions.php]: Fixed

File: functions.php (hunk 5 above)
Comment (mgallagher56) [reply functions.php]: Fixed

File: functions.php (hunk 9 above)
Comment (mgallagher56) [reply functions.php]: Fixed

File: functions.php (hunk 10 above)
Comment (mgallagher56) [reply functions.php]: Fixed

File: functions.php (hunk 11 above)
Comment (mgallagher56) [reply functions.php]: fixed

File: index.php (hunk 6 above)
Comment (mgallagher56) [reply index.php]: Fixed

File: index.php (hunk 7 above)
Comment (mgallagher56) [reply index.php]: Fixed

File: marcCollection_2019-09-23.sql (hunk 8 above)
Comment (mgallagher56) [reply marcCollection_2019-09-23.sql]: Fixed

**Review by charliecog — APPROVED** [review]
Nice work!

**Pull request merged**